src/NuGet.Core/NuGet.Commands/LocalsCommandRunner.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Common;
+using NuGet.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace NuGet.Commands
+{
+    public class LocalsCommandRunner
+    {
+        private enum LocalResourceName
+        {
+            Unknown,
+            HttpCache,
+            GlobalPackagesFolder,
+            Temp,
+            All
+        }
+        /// <summary>
+        /// Enum used to indicate the result of the command to callers.
+        /// </summary>
+        public enum LocalsCommandResult
+        {
+            InvalidLocalResourceName,
+            ClearFailure,
+            ClearSuccess
+        }
+        private const string _httpCacheResourceName = "http-cache";
+        private const string _globalPackagesResourceName = "global-packages";
+        private const string _allResourceName = "all";
+        private const string _tempResourceName = "temp";
+
+        public LocalsCommandResult Result { get; private set; }
+
+        public bool Clear { get; set; }
+
+        public bool List { get; set; }
+
+        private IList<string> Arguments { get; set; }
+
+        private ISettings Settings { get; set; }
+
+        public LocalsCommandRunner(IList<string> arguments, ISettings settings, bool clear, bool list)
+        {
+            this.Arguments = arguments;
+            this.Settings = settings;
+            this.Clear = clear;
+            this.List = list;
+        }
+        /// <summary>
+        /// Executes the logic for nuget locals command.
+        /// </summary>
+        /// <returns></returns>
+        public void ExecuteCommand()
+        {
+            var localResourceName = GetLocalResourceName(Arguments[0]);
```

Comment:
Checks are done by callers - 
[LocalsCommand.cs](https://github.com/NuGet/NuGet.Client/blob/ecac235f6541a7673e544a157bb7b7d64cb91338/src/NuGet.Clients/NuGet.CommandLine/Commands/LocalsCommand.cs#L25)
[Xplat Locals Command](https://github.com/NuGet/NuGet.Client/blob/ecac235f6541a7673e544a157bb7b7d64cb91338/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/LocalsCommand.cs#L44)

Since both callers have a different way of handling errors, I have not consolidated this.